### PR TITLE
[DOC] Globals doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Note:  `English:` (with colon) does not generate a link, so I used `English -`,

### CODE REVIEW

File: doc/globals.rdoc [hunk 1]
```diff
@@ -1,69 +1,425 @@
...
+
+==== <tt>$\\</tt> (Output Record Separator)
+
+An output record separator, initially +nil+.
+
+English - <tt>$OUTPUT_RECORD_SEPARATOR</tt>, <tt>$ORS</tt>.
+
+==== <tt>$,</tt> (Output Field Separator)
+
+An output field separator, initially +nil+.
```
Comment (peterzhu2118) [inline doc/globals.rdoc:120]: We should keep the fact that this field is deprecated (and maybe move it to the "Deprecated" section). For example, running the following script with warnings enabled (`-w` flag):

```ruby
$, = "hello"

puts ["one", "two", "three"].join
```

Outputs:

```
test.rb:1: warning: `$,' is deprecated
test.rb:3: warning: $, is set to non-nil value
onehellotwohellothree
```

File: doc/globals.rdoc [hunk 2]
```diff
@@ -1,69 +1,425 @@
...
+
+  $stdout # => #<IO:<STDOUT>>
+
+==== <tt>$stderr</tt> (Standard Error)
+
+The current standard error stream; initially:
+
+  $stderr # => #<IO:<STDERR>>
+
+==== <tt>$</tt> (Default Standard Input)
```
Comment (peterzhu2118) [inline doc/globals.rdoc:144]: Do you mean `$<`? I'm also not sure if "Default Standard Input" is the right term for this. `ARGF` is used to operate on files passed as command-line arguments (or stdin if there are no files passed in).

File: doc/globals.rdoc [hunk 3]
```diff
@@ -1,69 +1,425 @@
...
+    raise RuntimeError.new('Boo!')
+  rescue RuntimeError
+    p $!
+  end
+
+Output:
+
+  #<RuntimeError: Boo!>
+
+English - <tt></tt>ERROR_INFO
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/globals.rdoc]: Fixed.

File: doc/globals.rdoc [hunk 4]
```diff
@@ -1,69 +1,425 @@
...
+  end
+
+Output:
+
+  ["(irb):338:in `<top (required)>'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/workspace.rb:119:in `eval'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/workspace.rb:119:in `evaluate'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/context.rb:502:in `evaluate'"]
+
+English - $ERROR_POSITION
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/globals.rdoc]: Fixed.

File: doc/globals.rdoc [hunk 5]
```diff
@@ -1,69 +1,425 @@
...
+==== <tt>$~</tt> (\MatchData)
+
+MatchData object created from the match;
+thread-local and frame-local.
+
+English - <tt>$LAST_MATCH_INFO</tt>.
+
+==== <tt>$&</tt> (Matched Substring)
+
+The matchedstring.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply doc/globals.rdoc]: Fixed.

File: doc/globals.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/globals.rdoc]: Fixed and moved.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**